components/motor/gpio/basic.go

```diff
@@ -328,7 +329,25 @@ func (m *Motor) GoTo(ctx context.Context, rpm, positionRevolutions float64, extr
 
 // SetRPM instructs the motor to move at the specified RPM indefinitely.
 func (m *Motor) SetRPM(ctx context.Context, rpm float64, extra map[string]interface{}) error {
-	return motor.NewSetRPMUnsupportedError(m.Name().ShortName())
+	if m.maxRPM == 0 {
+		return errors.New("not supported, define max_rpm attribute != 0")
+	}
+
+	warning, err := checkSpeed(rpm, m.maxRPM)
+	if warning != "" {
+		m.logger.CWarnf(ctx, warning)
+	}
+	if err != nil {
+		return err
+	}
+
+	powerPct := rpm / m.maxRPM
```

Comment:
doesn't seem like this is a linear relationship, the calculated powerPct does not result in the input rpm. might try to look into a better way to do this but it's unencoded so there's not any good way to figure out what the motor is doing.

Reply:
it is unfortunately not. 
![image](https://github.com/viamrobotics/rdk/assets/35934754/5f8a529e-6a7c-4058-8290-7cd64bcef862)

Reply:
mm yeah alright that's a bummer